### PR TITLE
Adding a PhantomData example.

### PR DESCRIPTION
**Pull request by golddranks**
There was a FIXME in the PhantomData documentation, demanding more/better examples. I have recently used PhantomData this way, so I thought this could maybe do as an example.

r? @steveklabnik

### CODE REVIEW

File: src/libcore/marker.rs [hunk 1]
```diff
@@ -351,7 +351,35 @@ pub trait PhantomFn<A:?Sized,R:?Sized=()> { }
 /// instance, it will behave *as if* an instance of the type `T` were
 /// present for the purpose of various automatic analyses.
 ///
-/// For example, embedding a `PhantomData<T>` will inform the compiler
+/// # Example
```
Comment (steveklabnik) [inline src/libcore/marker.rs]: We use 'Examples' for everything, even if there's just one. And, there needs to be a newline after it as well.

File: src/libcore/marker.rs [hunk 2]
```diff
@@ -351,7 +351,35 @@ pub trait PhantomFn<A:?Sized,R:?Sized=()> { }
...
+/// When handling external resources over a foreign function interface,
+/// PhantomData can prevent mismatches by enforcing types in
```
Comment (steveklabnik) [inline src/libcore/marker.rs]: The type name should go in `s.

File: src/libcore/marker.rs [hunk 3]
```diff
@@ -351,7 +351,35 @@ pub trait PhantomFn<A:?Sized,R:?Sized=()> { }
...
+/// the method implementations, although the struct doesn't actually
+/// contain values of the resource type.
+///
+///```
+///pub struct ExternalResource<R> {
```
Comment (steveklabnik) [inline src/libcore/marker.rs]: Does this pass the tests? I thought (though I could be wrong) that using `pub` like this causes a test failure.

File: src/libcore/marker.rs (hunk 2 above)
Comment (steveklabnik) [inline src/libcore/marker.rs]: and it's `PhantomData<T>`.

**Comment by steveklabnik** [comment]
Hey @golddranks , thank you for the PR! I have a few nits, but I'm happy with this overall. @nikomatsakis , what do you think?

File: src/libcore/marker.rs (hunk 2 above)
Comment (golddranks) [inline src/libcore/marker.rs]: Fix'd, pushing in a sec.

File: src/libcore/marker.rs (hunk 3 above)
Comment (golddranks) [inline src/libcore/marker.rs]: Ah, I tested this on a fresh cargo project, and took the relevant part. Is there any in-tree-tests that should be implemented or ...?

**Comment by nikomatsakis** [comment]
@bors r+ 6eb5a37

lgtm!

**Comment by Manishearth** [comment]
This doesn't pass doctests

http://buildbot.rust-lang.org/builders/auto-mac-32-opt/builds/3982/steps/test/logs/stdio

Could you ensure that `make check-stage1-doc-crate-core` passes?

**Comment by golddranks** [comment]
Said this on IRC:
< drasa> steveklabnik: Ah, sorry, I just noticed you pulled in the PR of PhantomData
               example, I was actually unaware that code examples in the comments were
               tested too, so I hadn't added the "invisible" lines so that the tests would
               pass. I just learned that today, and was going to fix it, sorry for the
               trouble!

Closing this :)

**Pull request closed without merging**